Mage.Tests/src/test/java/org/mage/test/cards/single/afc/ShareTheSpoilsTest.java

```diff
@@ -0,0 +1,483 @@
+
+        skipInitShuffling();
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, shareTheSpoils);
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Tana, the Bloodsower");
+
+        setStopAt(1, PhaseStep.END_TURN);
+
+        execute();
+        assertAllCommandsUsed();
+
+        assertPermanentCount(playerA, "Tana, the Bloodsower", 1);
+
+        assertExileCount(playerA, "Tana, the Bloodsower", 0);
+        assertExileCount(playerA, "Reliquary Tower", 1);
+
+        assertExileCount(playerA, 1);
+        assertExileCount(playerB, 1);
+        assertExileCount(playerC, 1);
+        assertExileCount(playerD, 1);
+    }
+
+    /**
+     * During your turn, you may play A land from the cards exiled with share the spoils.
+     */
+    @Test
+    public void playLandOnOwnTurn() {
+        addCard(Zone.HAND, playerA, shareTheSpoils);
+        addCard(Zone.BATTLEFIELD, playerA, "Mountain", 8);
+
+        // 3rd from the top, exiled when Exotic Orchard is played with Share the Spoils
+        addCard(Zone.LIBRARY, playerA, "Reliquary Tower");
+        // 2nd from the top, exile when Share the Spoils is cast
+        addCard(Zone.LIBRARY, playerA, "Exotic Orchard");
+        // Topmost, draw at beginning of turn
+        addCard(Zone.LIBRARY, playerA, "Tana, the Bloodsower", 1); // {2}{R}{G}
+
+        skipInitShuffling();
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, shareTheSpoils);
+        playLand(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Exotic Orchard");
+
+        setStopAt(1, PhaseStep.END_TURN);
+
+        execute();
+        assertAllCommandsUsed();
+
+        assertPermanentCount(playerA, "Exotic Orchard", 1);
+
+        assertExileCount(playerA, "Exotic Orchard", 0);
+        assertExileCount(playerA, "Reliquary Tower", 1);
+
+        assertExileCount(playerA, 1);
+        assertExileCount(playerB, 1);
+        assertExileCount(playerC, 1);
+        assertExileCount(playerD, 1);
+    }
+
+    /**
+     * You cannot play a card or cast a spell when it's not your turn.
+     */
+    @Test
+    public void cannotCastWhenNotYourTurn() {
+        addCard(Zone.HAND, playerA, shareTheSpoils);
+        addCard(Zone.BATTLEFIELD, playerA, "Mountain", 8);
+
+        // 2nd from the top, exile when Share the Spoils is cast
+        addCard(Zone.LIBRARY, playerA, "Lightning Bolt", 1); // {R}
+        // Topmost, draw at beginning of turn
+        addCard(Zone.LIBRARY, playerA, "Exotic Orchard");
+
+        addCard(Zone.LIBRARY, playerB, "Reliquary Tower");
+
+        skipInitShuffling();
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, shareTheSpoils);
+        checkPlayableAbility("normal cast", 2, PhaseStep.PRECOMBAT_MAIN, playerA, "Lightning Bolt", false);
+        // TODO: How to do this properly for land?
+        checkPlayableAbility("normal cast", 2, PhaseStep.PRECOMBAT_MAIN, playerA, "Reliquary Tower", false);
+
+        setStopAt(2, PhaseStep.END_TURN);
+
+        execute();
```

Comment:
All new tests must contains `setStrictChooseMode(true)` before execute call. It helps to make sure that the command sequence is correct (you will see error on wrong or missing command).